File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        if (strs.length == 0) {
            return new ArrayList<>();
        }
        
        Map<String, List<String>> map = new HashMap<>();
        
        for (String i : strs) {
            char[] ch = i.toCharArray();
            Arrays.sort(ch);
            String sortedStr = String.valueOf(ch);
            
            if (!map.containsKey(sortedStr)) {
                map.put(sortedStr, new ArrayList<>());
            }
            
            map.get(sortedStr).add(i);
        }
        
        return new ArrayList<>(map.values());
    }
}

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        mp=defaultdict(list)
        for s in strs:
            freq=[0]*26
            for ch in s:
                freq[ord(ch)-ord('a')]+=1
            mp[tuple(freq)].append(s)
        return list(mp.values())